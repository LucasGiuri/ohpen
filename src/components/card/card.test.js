import Card from './card';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

describe('Card', () => {
  const title = 'title';
  const dataTestPrefix = 'dataTestPrefix';
  const content = 'customContent';
  const onCheck = jest.fn();

  describe('Snapshots', () => {
    it('default snapshot render', () => {
      const tree = renderer.create(<Card title={title} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with custom dataTestPrefix snapshot render', () => {
      const tree = renderer.create(<Card title={title} dataTestPrefix={dataTestPrefix} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with isChecked = true snapshot render', () => {
      const tree = renderer.create(<Card title={title} isChecked />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with isChecked = false snapshot render', () => {
      const tree = renderer.create(<Card title={title} isChecked={false} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with custom content snapshot render', () => {
      const tree = renderer.create(<Card title={title} content={content} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
