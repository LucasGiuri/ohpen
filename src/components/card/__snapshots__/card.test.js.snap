// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots default snapshot render 1`] = `
<div
  className="sc-bdfBQB fINfZT"
  data-test="card-card"
>
  <div
    className="sc-gsTEea bMmgDr"
  >
    <span
      className="sc-dlfnuX cvZFiX"
      data-test="card-card-title"
    >
      TITLE
    </span>
    <div
      className="sc-eCstlR jXyqPy"
    >
      <input
        className="sc-gKseQn gLUVYz"
        type="checkbox"
      />
      <label
        className="sc-jSgvzq iZDWJJ"
        htmlFor="checkbox"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-hKgJUU dbFtMW"
    data-test="card-card-content"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit
  </div>
</div>
`;

exports[`Card Snapshots with custom content snapshot render 1`] = `
<div
  className="sc-bdfBQB fINfZT"
  data-test="card-card"
>
  <div
    className="sc-gsTEea bMmgDr"
  >
    <span
      className="sc-dlfnuX cvZFiX"
      data-test="card-card-title"
    >
      TITLE
    </span>
    <div
      className="sc-eCstlR jXyqPy"
    >
      <input
        className="sc-gKseQn gLUVYz"
        type="checkbox"
      />
      <label
        className="sc-jSgvzq iZDWJJ"
        htmlFor="checkbox"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-hKgJUU dbFtMW"
    data-test="card-card-content"
  >
    customContent
  </div>
</div>
`;

exports[`Card Snapshots with custom dataTestPrefix snapshot render 1`] = `
<div
  className="sc-bdfBQB fINfZT"
  data-test="dataTestPrefix-card"
>
  <div
    className="sc-gsTEea bMmgDr"
  >
    <span
      className="sc-dlfnuX cvZFiX"
      data-test="dataTestPrefix-card-title"
    >
      TITLE
    </span>
    <div
      className="sc-eCstlR jXyqPy"
    >
      <input
        className="sc-gKseQn gLUVYz"
        type="checkbox"
      />
      <label
        className="sc-jSgvzq iZDWJJ"
        htmlFor="checkbox"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-hKgJUU dbFtMW"
    data-test="dataTestPrefix-card-content"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit
  </div>
</div>
`;

exports[`Card Snapshots with isChecked = false snapshot render 1`] = `
<div
  className="sc-bdfBQB fINfZT"
  data-test="card-card"
>
  <div
    className="sc-gsTEea bMmgDr"
  >
    <span
      className="sc-dlfnuX cvZFiX"
      data-test="card-card-title"
    >
      TITLE
    </span>
    <div
      className="sc-eCstlR jXyqPy"
    >
      <input
        className="sc-gKseQn gLUVYz"
        type="checkbox"
      />
      <label
        className="sc-jSgvzq iZDWJJ"
        htmlFor="checkbox"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-hKgJUU dbFtMW"
    data-test="card-card-content"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit
  </div>
</div>
`;

exports[`Card Snapshots with isChecked = true snapshot render 1`] = `
<div
  className="sc-bdfBQB fINfZT"
  data-test="card-card"
>
  <div
    className="sc-gsTEea bMmgDr"
  >
    <span
      className="sc-dlfnuX cvZFiX"
      data-test="card-card-title"
    >
      TITLE
    </span>
    <div
      className="sc-eCstlR jXyqPy"
    >
      <input
        className="sc-gKseQn gLUVYz"
        type="checkbox"
      />
      <label
        className="sc-jSgvzq iZDWJJ"
        htmlFor="checkbox"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-hKgJUU dbFtMW"
    data-test="card-card-content"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit
  </div>
</div>
`;
